src/SIL.Harmony/Db/CrdtRepository.cs

```diff
@@ -36,6 +33,11 @@ public Task<IDbContextTransaction> BeginTransactionAsync()
         return _dbContext.Database.BeginTransactionAsync();
     }
 
+    public SnapshotScope SnapshotScope()
+    {
+        return new SnapshotScope(this);
```

Comment:
I really don't like having state on the repo so I reworked this, it also turns out to be much simpler to see what's going on now that we don't have that callback being passed in